Automattic/woocommerce-payments
# Use Docker container for running Playwright e2e tests locally to make visual comparisons more robust

#### Changes proposed in this Pull Request

Based on #7671 – this PR changes the commands for running Playwright locally (`npm run test:e2e-pw`) to now use the official Playwright Docker container.

This means visual comparison tests are generated from a Linux container when running locally on macOS, ensuring consistent font-rendering.

> [!IMPORTANT]  
> This PR changes the e2e client's `wp-config.php` to make the WP hostname dynamic, which is necessary for Playwright to access the e2e client WP site with the URL `host.docker.internal`.

#### Testing instructions


* Test Playwright headless mode with `npm run test:e2e-pw`
* Test Playwright UI mode with `npm run test:e2e-pw-ui`
* Ensure existing e2e Puppeteer tests are not impacted. I've run the e2e tests workflow against this branch [here](https://github.com/Automattic/woocommerce-payments/actions/runs/7840551041)

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. **N/A, merging to feature branch**
